src/Zynga/Framework/StorableObject/Collections/Map/V2/Base.hh

```diff
@@ -0,0 +1,188 @@
+<?hh // strict
+
+namespace Zynga\Framework\StorableObject\Collections\Map\V2;
+
+use
+  Zynga\Framework\StorableObject\Collections\Map\V2\Exporter\Base as Exporter
+;
+use
+  Zynga\Framework\StorableObject\Collections\Map\V2\Importer\Box as BoxImporter
+;
+use
+  Zynga\Framework\StorableObject\Collections\Map\V2\Importer\Storable as StorableImporter
+;
+use
+  Zynga\Framework\StorableObject\Collections\V2\Interfaces\StorableMapCollection
+;
+use
+  Zynga\Framework\StorableObject\V1\Exceptions\OperationNotSupportedException
+;
+use Zynga\Framework\StorableObject\V1\Fields;
+use Zynga\Framework\StorableObject\V1\Fields\Generic as FieldsGeneric;
+use Zynga\Framework\StorableObject\V1\Interfaces\ExportInterface;
+use Zynga\Framework\StorableObject\V1\Interfaces\FieldsInterface;
+use Zynga\Framework\StorableObject\V1\Interfaces\ImportInterface;
+use Zynga\Framework\StorableObject\V1\Interfaces\StorableObjectInterface;
+use Zynga\Framework\Type\V1\Interfaces\TypeInterface;
+
+class Base<Tv> implements StorableMapCollection<Tv> {
```

Comment:
I know you are following convention, but can we change the class name to 'Map' here? I think the 'Base' naming convention may end up being broken in some cases like this. Also, we just need to revisit naming and grouping of classes as a whole but that can be done later.

```
class Map<Tv> implements StorableMapCollection<Tv> {
//....
```

Reply:
I am keeping this as it is, I generally don't like to repeat the path again. Collections/Map/V2/Map.hh anyways. 
As you said we can tackle grouping and naming later.